apps/website/src/components/admin/short-links/ShortLink.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState } from "react";
+import Link from "next/link";
+import type { inferRouterOutputs } from "@trpc/server";
+import { trpc } from "@/utils/trpc";
+import {
+  Button,
+  dangerButtonClasses,
+  LinkButton,
+} from "@/components/shared/Button";
+import { ModalDialog } from "@/components/shared/ModalDialog";
+import { Headline } from "@/components/admin/Headline";
+import { Panel } from "@/components/admin/Panel";
+import IconPencil from "@/icons/IconPencil";
+import IconTrash from "@/icons/IconTrash";
+import type { AppRouter } from "@/server/trpc/router/_app";
+type RouterOutput = inferRouterOutputs<AppRouter>;
+type ShortLink = RouterOutput["adminShortLinks"]["getLinks"][number];
+
+type LinkProps = {
+  shortLink: ShortLink;
+  onError: (error: string) => void;
+  onUpdate: () => void;
+};
+
+function ShortLinks({ shortLink, onError, onUpdate }: LinkProps) {
+  const deleteMutation = trpc.adminShortLinks.deleteLink.useMutation({
+    onError: (error) => onError(error.message),
+    onSettled: () => onUpdate(),
+  });
+  const clicks = trpc.adminShortLinks.getClicks.useQuery();
+
+  return (
+    <>
+      <tr className="border-b border-gray-700">
+        <td className="p-1">
+          <Button size="small" width="auto">
+            {"✅"}
+          </Button>
+        </td>
+        <td className="w-1/2 p-1">
+          <div className="flex flex-col gap-0.5">
+            <div className="text-xl">{shortLink.label}</div>
+            <div className="flex flex-col gap-1">
+              <Link
+                className="underline"
+                href={`/l/${shortLink.slug || shortLink.id}`}
+                target="_blank"
+              >
+                Public Link: {shortLink.slug || shortLink.id}
+              </Link>
+            </div>
+          </div>
+        </td>
+        <td className="">
+          <Link className="underline" href={shortLink.link} target="_blank">
+            {shortLink.link}
+          </Link>
+        </td>
+        <td>
+          {clicks.data?.map((c) =>
+            shortLink.id === c.id ? String(c.clicks) : "",
+          )}
```

Comment:
This feels rather inefficient -- can we do a reduce over `clicks.data` once at the start to construct a `Map`, so we can do an O(1) lookup here, rather than a map for every row?